src/components/dashboards/Admin-dashboard.tsx

```diff
@@ -63,40 +63,39 @@ export default function AdminDashboard() {
     query: string,
     filters: Record<string, boolean>
   ) => {
-    //TODO: Implement search and filter
-    if (query && query.length > 0) {
-      setIsSearch(true);
-    }
     try {
-      const searchResults = await fetchSearchResults(query, filters);
-      if (searchResults?.residentRequests) {
-        setUserRequests(searchResults.residentRequests);
-        setTotalCount(searchResults.count);
-      }
-    } catch (error) {
-      console.log("error fetching results", error);
-    }
-  };
+      const allRequests = await ResidentRequestService.adminGetAllRequests({
+        take: MAX_TAKE,
+        skip: pageNumber * MAX_TAKE,
+      });
+
+      let filteredRequests = allRequests.residentRequests;
 
-  const fetchSearchResults = useCallback(
-    debounce(async (query: string, filters: Record<string, boolean>) => {
-      if (!query) {
-        setUserRequests([]);
-        return;
+      if (!filters.all) {
+        filteredRequests = filteredRequests.filter(
+          (request) =>
+            (filters.completed && request.status === "COMPLETED") ||
+            (filters.pending && request.status === "PENDING") ||
+            (filters.canceled && request.status === "CANCELED")
```

Comment:
Use enum instead of hard-coded string
```suggestion
            (filters.completed && request.status === RequestStatus.Completed) ||
            (filters.pending && request.status === RequestStatus.Pending) ||
            (filters.canceled && request.status === RequestStatus.Canceled)
```